### PR TITLE
Add passphrase parameter when creating a wallet

### PR DESCRIPTION
**Pull request by OBorce**
Add optional passphrase parameter to the wallet commands

### CODE REVIEW

File: wallet/wallet-rpc-client/src/handles_client/mod.rs [hunk 1]
```diff
@@ -103,14 +105,15 @@ impl<N: NodeInterface + Clone + Send + Sync + 'static + Debug> WalletInterface
         path: PathBuf,
         store_seed_phrase: bool,
         mnemonic: Option<String>,
+        passphrase: Option<String>,
     ) -> Result<CreatedWallet, Self::Error> {
```
Comment (TheQuantumPhysicist) [inline wallet/wallet-rpc-client/src/handles_client/mod.rs:109]: This doesn't model the way the wallet work correctly. The case with mnemonic being None and passphrase being Some is possible and this leads to having to manually check this or expect on it, unnecessarily. This should be enforced at the compiler level. Despite being ugly, something like `Option<(String, Option<String>)>` is way more appropriate (with docs explaining it). A struct for this is also OK.

**Review by TheQuantumPhysicist — APPROVED** [review]
Besides the parametrization, this looks OK.

Though I would recommend bumping the database version, even though no upgrade is required. Make an upgrade and make it empty. This is because otherwise users could try to open the wallet with an older version and will get an error.

File: wallet/wallet-rpc-client/src/handles_client/mod.rs (hunk 1 above)
Comment (OBorce) [reply wallet/wallet-rpc-client/src/handles_client/mod.rs:109]: I thought the user might want to have a random seed phrase, but have his own passphrase

Comment (TheQuantumPhysicist) [reply wallet/wallet-rpc-client/src/handles_client/mod.rs:109]: Actually you're right. This is a valid case. Please document the meaning of these options, if you haven't yet.

File: wallet/wallet-cli-lib/src/commands/command_handler/mod.rs [hunk 2]
```diff
@@ -234,8 +260,12 @@ where
             ColdWalletCommand::PurgeSeedPhrase => {
                 let phrase = self.non_empty_wallet().await?.purge_seed_phrase().await?;
 
-                let msg = if let Some(phrase) = phrase.seed_phrase {
-                    format!("The seed phrase has been deleted, you can store it if you haven't do so yet: \"{}\"", phrase.join(" "))
+                let msg = if let Some(phrase) = phrase {
+                    if let Some(passphrase) = phrase.passphrase {
+                        format!("The seed phrase has been deleted, you can store it if you haven't do so yet: \"{}\" with passphrase \"{}\"", phrase.seed_phrase.join(" "), passphrase)
```
Comment (TheQuantumPhysicist) [inline wallet/wallet-cli-lib/src/commands/command_handler/mod.rs:265]: "haven't done so yet"

Comment (TheQuantumPhysicist) [reply wallet/wallet-cli-lib/src/commands/command_handler/mod.rs:265]: Also, in separate lines, please.

**Comment by OBorce** [comment]
will create a new PR with the review comments

**Pull request merged**